### PR TITLE
feat: signPSBT

### PR DESCRIPTION
**Pull request by rolznz**
- [x] use new bitcoin object and remove added lib functions. Use new liquid secp provider for signing
- [x] remove experimental webpack flag
- [x] review addition of new libraries
- [ ] review inconsistency with liquid - here signPsbt actually uses hex, liquid uses base64. Hex seems more accepted for bitcoin tools and APIs.

### Describe the changes you have made in this PR

Ability to sign partially signed bitcoin transactions

Test it at: https://getalby.github.io/bitcoin-wallet/

### Type of change

(Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

image

image

_Add screenshots to make your changes easier to understand. You can also add a video here._

### How has this been tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration_

### Checklist

- [x] Self-review of changed code
- [x] Manual testing
- [x] Added automated tests where applicable
- [x] Update Docs & Guides
- For UI-related changes
- [x] Darkmode
- [x] Responsive layout

### CODE REVIEW

File: src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  const navigate = useNavigate();
+
+  const psbt = navState.args?.psbt as string;
+  const origin = navState.origin as OriginData;
+  const [loading, setLoading] = useState(false);
+  const [successMessage, setSuccessMessage] = useState("");
+  const [preview, setPreview] = useState<PsbtPreview | undefined>(undefined);
+  const [showAddresses, setShowAddresses] = useState(false);
+  const [showHex, setShowHex] = useState(false);
+  const [error, setError] = useState<string | undefined>();
```
Comment (rolznz) [inline src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx:34]: Don't need a separate error state. Just show the toast?

File: src/extension/content-script/webbtc.js [hunk 2]
```diff
@@ -24,6 +25,16 @@ async function init() {
     return;
   }
 
+  browser.runtime.onMessage.addListener((request, sender, sendResponse) => {
+    // forward account changed messaged to inpage script
+    if (request.action === "accountChanged") {
```
Comment (bumi) [inline src/extension/content-script/webbtc.js:30]:
```suggestion
    if (request.action === "accountChanged" && isEnabled) {
```

Comment (rolznz) [reply src/extension/content-script/webbtc.js:30]: Make sure it is this way for all providers

[the remark this answers is not in the document]
Comment (reneaaron) [reply src/extension/content-script/webbtc.js:30]: Also make sure this is applied to other provider's contentscripts

File: src/i18n/locales/en/translation.json [hunk 3]
```diff
@@ -818,6 +818,18 @@
...
+        "title": "Sign",
+        "allow_sign": "This website asks you to sign a Bitcoin Transaction:",
+        "view_details": "View details",
+        "hide_details": "Hide details",
+        "view_raw_transaction": "View raw transaction (Hex)",
+        "hide_raw_transaction": "Hide raw transaction (Hex)",
+        "inputs": "Inputs",
+        "outputs": "Outputs",
+        "fee": "Fee",
+        "amount": "{{amount}} sats"
```
Comment (reneaaron) [inline src/i18n/locales/en/translation.json:831]: Use `getFormattedSats()` instead, remove translation.

File: src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+              )}
+            </div>
+
+            {showHex && (
+              <div className="break-all p-2 mb-4 shadow bg-white rounded-lg dark:bg-surface-02dp text-gray-500 dark:text-gray-400">
+                {psbt}
+              </div>
+            )}
+          </div>
+          <ConfirmOrCancel
```
Comment (rolznz) [inline src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx:152]: There is no margin below, can this be fixed?

File: src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                  <Hyperlink onClick={toggleShowHex}>
+                    {showHex
+                      ? t("hide_raw_transaction")
+                      : t("view_raw_transaction")}
+                  </Hyperlink>
+                </>
+              )}
+            </div>
+
+            {showHex && (
```
Comment (rolznz) [inline src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx:146]: Don't put this in a separate container

Comment (rolznz) [reply src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx:146]: Make it work consistent with Nostr (centered View details)

File: src/i18n/locales/en/translation.json [hunk 6]
```diff
@@ -818,6 +818,18 @@
       "confirm_get_address": {
         "title": "Get Address"
       },
+      "confirm_sign_psbt": {
+        "title": "Sign",
+        "allow_sign": "This website asks you to sign a Bitcoin Transaction:",
```
Comment (rolznz) [inline src/i18n/locales/en/translation.json:823]: remove semicolon

Comment (rolznz) [reply src/i18n/locales/en/translation.json:823]: Make bitcoin and transaction lowercase

**Review by reneaaron — COMMENTED** [review]
Nice work! :rocket:

File: src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx (hunk 4 above)
Comment (rolznz) [reply src/app/screens/Bitcoin/ConfirmSignPsbt/index.tsx:152]: this was fixed after merging master

File: src/i18n/locales/en/translation.json [hunk 7]
```diff
@@ -827,7 +827,7 @@
     },
     "confirm_sign_pset": {
       "title": "Sign",
-      "allow_sign": "This website asks you to sign a Liquid Transaction:",
+      "allow_sign": "This website asks you to sign a liquid transaction",
```
Comment (reneaaron) [inline src/i18n/locales/en/translation.json:830]:
```suggestion
      "allow_sign": "This website asks you to sign a Liquid transaction",
```

**Review by reneaaron — APPROVED** [review]
tACK

**Pull request merged**